### ISSUE TITLE
Type error after upgrading to webpack@5.94.0 (A class can only implement an object type or intersection of object types with statically known members.)

### ISSUE THREAD

**Issue by rostislav-simonik** [dividab/tsconfig-paths-webpack-plugin#109]
```
export declare class TsconfigPathsPlugin implements ResolvePluginInstance {
    source: string;
    target: string;
    log: Logger.Logger;
    baseUrl?: string | undefined;
    absoluteBaseUrl: string;
    extensions: ReadonlyArray<string>;
    referenceMatchMap: Record<string, TsconfigPaths.MatchPathAsync>;
    matchPath: TsconfigPaths.MatchPathAsync;
    constructor(rawOptions?: Partial<Options.Options>);
    apply(resolver: Resolver): void;
}
```
is failing on `implements ResolvePluginInstance` with error 
```
A class can only implement an object type or intersection of object types with statically known members.ts(2422)
(alias) type ResolvePluginInstance = {
    [index: string]: any;
    apply: (arg0: Resolver) => void;
} | ((this: Resolver, arg1: Resolver) => void)
import ResolvePluginInstance
```

**Comment by Yurickh** [comment]
Looks like it has become incompatible due to this change in the `ResolvePluginInstance` type in webpack:
https://github.com/webpack/webpack/commit/cf891a416465c5a57201a4fbffb2e6eb114e96dd

**Comment by kvndrsslr** [comment]
@jonaskello Would you be able to release a patch with this change anytime soon?

**Comment by jonaskello** [comment]
This is now published in 4.2.0
